### ISSUE TITLE
Arguments for ParamSpec is missing

### ISSUE THREAD

**Issue by dbrattli** [microsoft/pyright#3482]
**Describe the bug**

Pyright seems to be unsure about the code and sometimes flags it as bug, sometimes not. I only see this error in vscode, never when running pyright from the command line. If I edit the code then the error will go away and then come back. So I'm unsure if I have made a mistake here or if this is an issue with pyright?

<img width="711" alt="Screenshot 2022-05-19 at 08 37 53" src="https://user-images.githubusercontent.com/849479/169226768-8ac4f8a1-eed5-47b5-9b20-f9d40b1b455d.png">

**To Reproduce**
Steps to reproduce the behavior. See repro code below:

**Expected behavior**

Expect pyright to be consistent if this code should type check or not. Currently it looks like it cannot really decide. Sometimes it's ok, sometimes not. Casting `bouncer` e.g `bouncer_ = cast(TailCall[_P], bouncer)` and use `_bouncer` instead does not help. I would appreciate any advice if my use of ParamSpec is wrong.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
import functools
from typing import Callable, Generic, TypeVar, Union

from typing_extensions import ParamSpec

_TResult = TypeVar("_TResult")
_P = ParamSpec("_P")


class TailCall(Generic[_P]):
    def __init__(self, *args: _P.args, **kw: _P.kwargs):
        self.args = args
        self.kw = kw


TailCallResult = Union[_TResult, TailCall[_P]]


def tailrec(fn: Callable[_P, TailCallResult[_TResult, _P]]) -> Callable[_P, _TResult]:
    def trampoline(bouncer: TailCallResult[_TResult, _P]) -> _TResult:
        while isinstance(bouncer, TailCall):
            args, kw = bouncer.args, bouncer.kw
            bouncer = fn(*args, **kw)

        return bouncer

    @functools.wraps(fn)
    def wrapper(*args: _P.args, **kw: _P.kwargs) -> _TResult:
        return trampoline(fn(*args, **kw))

    return wrapper
```

If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

**VS Code extension or command-line**
Pylance: v2022.5.2, Pyright 1.1.247

**Additional context**

This issue was not introduced by 1.1.247. I've seen this for quite some time.

**Comment by erictraut** [comment]
Thanks for the bug report. This occurs only in the case where a call expression is invoking a generic function that is parameterized with a ParamSpec and is located within a loop construct. And even then, it happens only in certain evaluation orderings. That explains why this wasn't reported previously.

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.248, which I just published. It will also be included in the next release of pylance.
